presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -447,6 +444,26 @@ public PlanWithProperties visitSort(SortNode node, PreferredProperties preferred
                 }
             }
 
+            if (isDistributedSortEnabled(session)) {
+                child = planChild(node, PreferredProperties.any());
```

Comment:
If distributed sort is enabled then we don't want to prefer `PreferredProperties.undistributed()` properties as it might hurt source stage processing performance.